smtchecker.ts

```diff
@@ -3,7 +3,7 @@
 // The function runs an SMT solver on each query and adjusts the input for
 // another run.
 // Returns null if no solving is requested.
-function handleSMTQueries (inputJSON, outputJSON, solver) {
+export function handleSMTQueries (inputJSON: any, outputJSON: any, solver?: any) {
```

Comment:
I don't think solver is optional?

Reply:
In multiple test cases, the solver is not provided meaning the build will fail, so this has to be defined as nullable. These tests will have to be written in a way to take in the solver first.

```
test/smtchecker.ts:8:25 - error TS2554: Expected 3 arguments, but got 2.

8     st.equal(smtchecker.handleSMTQueries(input, output), null);
                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  smtchecker.ts:6:68
    6 export function handleSMTQueries (inputJSON: any, outputJSON: any, solver: any) {
                                                                         ~~~~~~~~~~~
    An argument for 'solver' was not provided.

test/smtchecker.ts:15:25 - error TS2554: Expected 3 arguments, but got 2.

15     st.equal(smtchecker.handleSMTQueries(input, output), null);
                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  smtchecker.ts:6:68
    6 export function handleSMTQueries (inputJSON: any, outputJSON: any, solver: any) {
                                                                         ~~~~~~~~~~~
    An argument for 'solver' was not provided.

test/smtchecker.ts:22:25 - error TS2554: Expected 3 arguments, but got 2.

22     st.equal(smtchecker.handleSMTQueries(input, output), null);
                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  smtchecker.ts:6:68
    6 export function handleSMTQueries (inputJSON: any, outputJSON: any, solver: any) {
                                                                         ~~~~~~~~~~~
    An argument for 'solver' was not provided.
```

Reply:
@leonardoalt is this by design or is all this fixed by #565 ?

Reply:
Ah, `handleSMTQueries` actually needs to be changed accordingly in that PR, taking the `solverFunction` (which is its `solver` atm and the solver config object)

Reply:
The tests that call `handleSMTQueries` don't really need a solver, I guess that was the idea, to then just let it fail

Reply:
@leonardoalt can you incorporate fixes into #565 to the issues encountered here?

Reply:
I think I fixed them there now. `handleSMTQueries` now takes 4 arguments: the same input and output json, plus the solverFunction and the solver object. I think those two should be acceptable as null